lib/buffer.js

```diff
@@ -303,6 +303,24 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
+Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+  if (byteOffset > 0x7fffffff)
+    byteOffset = 0x7fffffff;
+  else if (byteOffset < -0x80000000)
+    byteOffset = -0x80000000;
+  byteOffset >>= 0;
+
+  if (typeof val === 'string')
+    return binding.indexOfString(this, val, byteOffset);
+  if (val instanceof Buffer)
+    return binding.indexOfBuffer(this, val, byteOffset);
+  if (typeof val === 'number')
```

Comment:
or just truncate in some way on `val > 255`, which is what is done for byte values elsewhere such as in [`buffer.fill()`](https://github.com/iojs/io.js/blob/36a779560a0a4d3583feb216a46a64dc3f8e7498/src/node_buffer.cc#L362) .